### PR TITLE
set: implement GetRegionCode

### PR DESCRIPTION
**Pull request by makigumo**
following https://switchbrew.org/wiki/Settings_services#RegionCode

### CODE REVIEW

File: src/core/settings.cpp [hunk 1]
```diff
@@ -86,6 +86,7 @@ void LogSettings() {
     LogSetting("System_RngSeed", Settings::values.rng_seed.value_or(0));
     LogSetting("System_CurrentUser", Settings::values.current_user);
     LogSetting("System_LanguageIndex", Settings::values.language_index);
+    LogSetting("System RegionIndex", Settings::values.region_index);
```
Comment (FearlessTobi) [inline src/core/settings.cpp]: This is not following our naming conventions.

File: src/core/hle/service/set/set.cpp [hunk 2]
```diff
@@ -111,14 +111,22 @@ void SET::GetLanguageCode(Kernel::HLERequestContext& ctx) {
     rb.PushEnum(available_language_codes[Settings::values.language_index]);
 }
 
+void SET::GetRegionCode(Kernel::HLERequestContext& ctx) {
+    LOG_DEBUG(Service_SET, "called");
+
+    IPC::ResponseBuilder rb{ctx, 3};
+    rb.Push(RESULT_SUCCESS);
+    rb.Push(static_cast<s32>(Settings::values.region_index));
```
Comment (FearlessTobi) [inline src/core/hle/service/set/set.cpp]: I'm not exactly sure if this cast is necessary.

**Comment by makigumo** [comment]
Thx for the review, feedback addressed.

Not sure if any game is using this, but it seemed simple enough to try to implement.

**Comment by CosmicDreamsOfCode** [comment]
Animal Crossing is using it. Other games could use it too, AC is just the only one i know of.

**Comment by FearlessTobi** [comment]
@CosmicDreamsOfCode Please refrain from posting such things or you will be banned from our repo permanently.

**Comment by CosmicDreamsOfCode** [comment]
> @CosmicDreamsOfCode Please refrain from posting such things or you will be banned from our repo permanently.

Was just trying to help clear up what games use it but point taken. Won't post about it again.

**Comment by makigumo** [comment]
Rebased.

**Comment by MattKackles** [comment]
animal crossing uses this. I dumped it from eshop legitimately, its out so...

**Comment by kipukun** [comment]
Can confirm that Animal Crossing 2020 uses this. I was investigating when I found this PR. 
The game crashes with something in `vi` after using this build.

**Comment by bunnei** [comment]
LGTM, thank!

**Pull request merged**